.github/workflows/update-version.yml

```diff
@@ -22,14 +25,42 @@ jobs:
         run: |
           git config user.name devops
           git config user.email devops@runtimeverification.com
-      - name: 'Install Poetry'
-        uses: Gr1N/setup-poetry@v9
+      - name: 'Install Nix'
+        uses: cachix/install-nix-action@v31.2.0
+        with:
+          install_url: https://releases.nixos.org/nix/nix-2.24.12/install
+          extra_nix_config: |
+            substituters = http://cache.nixos.org https://cache.iog.io
+            trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=
+            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+      - uses: cachix/cachix-action@v16
+        with:
+          name: k-framework
+          authToken: ${{ secrets.CACHIX_PUBLIC_TOKEN }}
+      # note: we install the same version of `uv` as used by `uv2nix` in order to match the nix derivation
+      - name: 'Install uv'
+        run: |
+          UV_VERSION=$(curl -s https://raw.githubusercontent.com/pyproject-nix/uv2nix/$(cat deps/uv2nix)/pkgs/uv-bin/srcs.json | jq -r .version)
+          [[ "$UV_VERSION" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]
+          curl -LsSf https://astral.sh/uv/$UV_VERSION/install.sh | sh && uv --version
```

Comment:
This is supported by the `setup-uv` action I think: https://github.com/astral-sh/setup-uv?tab=readme-ov-file#install-a-specific-version